test/Test_PIE.ml

```diff
@@ -76,10 +76,79 @@ let abs_zero_initial_features () =
     (List.map [(-1) ; 3 ; 0 ; (-2) ; 6] ~f:(fun i -> [Value.Int i])))
   in Alcotest.(check string) "identical" "(>= x 0)" (cnf_opt_to_desc res)
 
+
+ let real_abs_job = Job.create_unlabeled
+  ~f:(fun [@warning "-8"] [ Value.Real x ] -> Value.Real (if Float.(x > 0.) then x else (-1. *. x)))
+  ~args:([ ("x", Type.REAL) ])
+  ~post:(fun inp res ->
+          match [@warning "-8"] inp , res with
+          | [ Value.Real x ], Ok (Value.Real y) -> Float.(equal x y))
+  ~features:[
+    ((fun [@warning "-8"] [Value.Real x] -> Float.(x > 0.)), "(> x 0.)")
+  ]
+  (List.map [(-1.4) ; 3.5 ; 0. ; (-2.6) ; 6.0] ~f:(fun i -> [Value.Real i]))
+
+ let real_abs_conflict_failure () =
+  let (res, _) = learnPreCond real_abs_job ~config:{ PIE.Config.default with
+                                                disable_synth = true }
+  in Alcotest.(check string) "identical" "false" (cnf_opt_to_desc res)
+
+ let real_abs_conflict_group () =
+  let res = List.to_string_map (conflictingTests real_abs_job)
+                               ~sep:"\n" ~f:conflicts_to_string
+  in Alcotest.(check string) "identical" "Pos:{[0.]} + Neg:{[-2.6] ; [-1.4]}" res
+
```

Comment:
Oh sorry, I think I wasn't clear.  You don't need to have a version of each of the existing tests, and in general that's not the right thing to do.  Those are each testing some particular capability or behavior of PIE.  You just want to add new tests that exercise your new capabilities/behaviors.  So in your case, it should suffice to have a version for abs where you provide sufficient features, and one where you have to synthesize features.   You don't need to re-test the behavior under 1-CNF, the behavior of the conflictingTests function, etc.

Reply:
Oh I see, I'll change that. Thank you!